### ISSUE TITLE
Auth setup script

### ISSUE THREAD

**Issue by BrianHicks** [CiscoCloud/microservices-infrastructure#65]
We should have a crypto setup script to automate all the key setup. Related to #45 and #46.

The usage would be something like this:

```
$ crypto-setup init
===== consul =====
----> encryption key
no encryption key found, generating into inventory/group_vars/all/all.yml

==== marathon ====
----> certificate
no certificate found, generate? [Y/n]
generating certificate...
password? []: *****
writing certificate to inventory/marathon.jks
writing password to inventory/group_vars/marathon/all.yml

----> user
use basic auth to connect? [Y/n]
username [admin]: *****
password: *****
writing user:pass to inventory/group_vars/marathon/all.yml
```

If an update is needed (password change, etc), the user can run `crypto-setup update COMPONENT [COMPONENT...]`

The following needs to be done in this script (from @stevendborrelli):

- [x] mesos leader/follower credentials
- [x] mesos `--authenticate` flag (if credential is set)
- [x] mesos `--authenticate_slaves` flag (if credential is set)
- [x] consul acl_master token
- [x] consul gossip key
- [x] marathon framework auth
- [x] lock down mesos credentials (`0600`)
- [x] ZK credentials
- [x] ZK superuser
- [x] single Mesos follower credential
- [x] CA generation
- [x] certificate generation for nginx
- [x] certificate generation for consul
- [x] output JSON format in setup script to pass to the extra vars argument

**Comment by stevendborrelli** [comment]
+1 we need to focus on making it easy to set up security.

**Comment by matjohn2** [comment]
Could this just 'happen' instead of being a script that needs running/input? Secure by default seems like an even better path for usability.

**Comment by BrianHicks** [comment]
We're doing it this way to avoid using default passwords. This will make it as easy as possible to set up security without leaking secrets.

**Comment by stevendborrelli** [comment]
When security is set up, these are the items that will need to be changed:
 
- [ ] Create Self-signed TLS certs (this script)
- [ ] mesos leader/follower credentials (this script)
- [ ] mesos `--authenticate` flag (if credential is set)
- [ ] mesos `--authenticate_slaves` flag (if credential is set)
- [ ] consul acl_master token (this script)
- [ ] consul gossip key (this script)
- [ ] consul acl_datacenter
- [ ] consul default acl_policy
- [ ] consul ssl configuration
- [ ] k/v auth in consul for nginx container. Create username:password combo in this script, and a role to upload to consul k/v. Let's make the password bcrypt using `htpasswd -B` if nginx supports it.
- [ ] create zookeeper credentials
- [ ] create zookeeper ACL on `/mesos` znode (in playbook if zk security is set)
- [ ] create zookeeper ACL on `/marathon` znode (in playbook if zk security is set)
- [ ] create nginx + consul-template container

**Comment by BrianHicks** [comment]
I've added the relevant items to the description up top. I'm also going to change this to be noninteractive for now. It'll generate and store nicely random internal passwords and keys just fine without any user input.

**Comment by keithchambers** [comment]
+1

**Comment by BrianHicks** [comment]
merged into 0.2-dev
